main.go

```diff
@@ -0,0 +1,188 @@
+		}
+		return cfg, nil
+	}
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+func main() {
+	klog.InitFlags(nil)
+
+	var port int
+	var kubeconfig string
+	var leaseLockName string
+	var leaseLockNamespace string
+	var id string
+
+	flag.IntVar(&port, "port", 8080, "port to listen")
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.StringVar(&id, "id", "", "the holder identity name")
+	flag.StringVar(&leaseLockName, "lease-lock-name", "", "the lease lock resource name")
+	flag.StringVar(&leaseLockNamespace, "lease-lock-namespace", "", "the lease lock resource namespace")
+	flag.Parse()
+
+	if id == "" {
+		klog.Fatal("unable to get id (missing id flag).")
+	}
+	if leaseLockName == "" {
+		klog.Fatal("unable to get lease lock resource name (missing lease-lock-name flag).")
+	}
+	if leaseLockNamespace == "" {
+		klog.Fatal("unable to get lease lock resource namespace (missing lease-lock-namespace flag).")
+	}
+
+	// leader election uses the Kubernetes API by writing to a
+	// lock object, which can be a LeaseLock object (preferred),
+	// a ConfigMap, or an Endpoints (deprecated) object.
+	// Conflicting writes are detected and each client handles those actions
+	// independently.
+	config, err := buildConfig(kubeconfig)
+	if err != nil {
+		klog.Fatal(err)
+	}
+	client := clientset.NewForConfigOrDie(config)
+
+	// use a Go context so we can tell the leaderelection code when we
+	// want to step down
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// listen for interrupts or the Linux SIGTERM signal and cancel
+	// our context, which the leader election code will observe and
+	// step down
+	stoppedCh := make(chan os.Signal, 1)
+	defer close(stoppedCh)
+	signal.Notify(stoppedCh, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-stoppedCh
+		klog.Info("Received termination, signaling shutdown")
+		cancel()
+	}()
+
+	// Setup any healthz checks we will want to use.
+	var checks []healthz.HealthChecker
+	var electionChecker *leaderelection.HealthzAdaptor
+	electionChecker = leaderelection.NewLeaderHealthzAdaptor(time.Second * 20)
+	checks = append(checks, electionChecker)
+
+	mux := http.NewServeMux()
+	healthz.InstallPathHandler(mux, "/healthz", checks...)
+
+	server := &http.Server{
+		Addr:    fmt.Sprintf(":%d", port),
+		Handler: mux,
+	}
+
+	go func() {
+		klog.Infof("Start listening to %d", port)
+
+		if err := server.ListenAndServe(); err != nil {
+			klog.Fatalf("Error starting server: %v", err)
+		}
+	}()
+
+	// complete your controller loop here
+	run := func(ctx context.Context) {
+		klog.Info("Controller loop...")
+
+		select {}
+	}
+
+	klog.Infof("Run with leader election")
+
+	// we use the Lease lock type since edits to Leases are less common
+	// and fewer objects in the cluster watch "all Leases".
+	lock := &resourcelock.LeaseLock{
+		LeaseMeta: metav1.ObjectMeta{
+			Name:      leaseLockName,
```

Comment:
Whether the lease lock needs to be created in advance?

Reply:
no